lib/rules/valid-api.js

```diff
@@ -68,17 +68,15 @@ function validatePropertyName(property, context) {
         });
     }
 
-    if (name.startsWith('data')) {
+    if (RESERVED_PUBLIC_PROPERTIES.has(name)) {
         context.report({
             node: property,
-            message: `Invalid public property "${name}". Properties starting with "data" are reserved properties.`,
+            message: `Invalid public property "${name}". This property name is a reserved property.`,
         });
-    }
-
-    if (RESERVED_PUBLIC_PROPERTIES.has(name)) {
+    } else if (/^data[A-Z]/.test(name)) {
         context.report({
             node: property,
-            message: `Invalid public property "${name}". This property name is a reserved property.`,
+            message: `Invalid public property "${name}". Properties starting with "data" are reserved properties.`,
```

Comment:
```suggestion
            message: `Ambiguous public property name "${name}". The corresponding content attribute name is of the form "data-*" whose value must be reflected to the component instance as "dataset.${name}".`,
```

Reply:
```suggestion
            message: `Invalid public property name "${name}". Properties starting with "data" correspond to data-* HTML attributes, which are not allowed.`,
```
I think @ekashida's suggestion is too verbose. 😛

Reply:
Either way works, but can we add the detailed explanation (for my own sake) as a comment somewhere?